pootle/apps/pootle_language/forms.py

```diff
@@ -135,3 +150,116 @@ def save(self):
+        choices=(
+            [("", "-----")]
+            + [(k, v)
+               for k, v
+               in STATES_MAP.items() if k != OBSOLETE]),
+        widget=forms.Select(
+            attrs={'class': 'js-select2 select2-language'}))
+    filter_tp = forms.ModelChoiceField(
+        label=_("Project"),
+        required=False,
+        queryset=TranslationProject.objects.none(),
+        widget=forms.Select(
+            attrs={'class': 'js-select2'}))
+    suggestions = forms.ModelMultipleChoiceField(
+        widget=TableSelectMultiple(
+            item_attrs=[
+                "unit_link",
+                "unit_state",
+                "unit",
+                "target_f",
+                "user",
+                "creation_time",
+                "project"]),
+        required=False,
+        queryset=Suggestion.objects.select_related(
+            "unit",
+            "unit__store",
+            "unit__store__translation_project",
+            "unit__store__translation_project__project").none())
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop("user")
+        self.language = kwargs.pop("language")
+        super(LanguageSuggestionAdminForm, self).__init__(*args, **kwargs)
+        self.fields["filter_suggester"].choices = self.filter_suggester_choices
+        self.fields["filter_tp"].queryset = self.filter_tp_qs
+        self.fields["suggestions"].queryset = self.suggestions_qs
+
+    @property
+    def filter_suggester_choices(self):
+        suggesters = list(
+            (username,
+             ("%s (%s)" % (fullname, username)
+              if fullname.strip()
+              else username))
+            for username, fullname
+            in sorted(self.language_team.users_with_suggestions))
+        return [("", "-----")] + suggesters
+
+    @property
+    def suggestions_qs(self):
+        return self.language_team.suggestions.select_related("unit", "user").filter(
+            unit__store__translation_project__project__disabled=False)
+
+    @property
+    def filter_tp_qs(self):
+        tps = self.language.translationproject_set.exclude(
+            project__disabled=True)
+        tps = tps.filter(
+            stores__unit__suggestion__state=SuggestionStates.PENDING)
+        return tps.order_by("project__code").distinct()
+
+    @property
+    def suggestions_to_save(self):
+        if not self.is_valid():
+            return []
+        return (
+            self.fields["suggestions"].queryset
+            if self.cleaned_data["select_all"]
+            else self.cleaned_data["suggestions"])
+
+    @property
+    def suggestions_review(self):
+        if not self.is_valid():
+            return
+        return review.get(Suggestion)(self.suggestions_to_save, self.user)
+
+    def save(self):
+        return (
+            self.suggestions_review.accept(
+                comment=self.cleaned_data["comment"])
+            if self.cleaned_data["actions"] == "accept"
```

Comment:
Thinking aloud this would better to be some constant. But I am not blocking on it.